### PR TITLE
Added weights to the functions of Sylo module

### PR DESCRIPTION
**Pull request by roy-powerfinance**
Added weights, comments and compexity for these function

### CODE REVIEW

**Review by hoani — APPROVED** [review]
LGTM, nice job 👍

**Review by cowboy-bebug — APPROVED** [review]
LGTM

File: crml/sylo/src/e2ee.rs [hunk 1]
```diff
@@ -58,6 +76,12 @@ decl_module! {
 			<PreKeyBundles<T>>::mutate((sender, device_id), |current_pkbs| current_pkbs.extend(pkbs));
 		}
 
+		/// Retrieve and remove the Prekey bundles of a given list of user accounts and devices
+		///
+		/// weight:
+		/// O(n * k) where n is the size of input ```wanted_pkbs```, and k is the number existing PKBS in the storage
```
Comment (jordy25519) [inline crml/sylo/src/e2ee.rs]: \`wanted_pkbs\`

File: crml/sylo/src/e2ee.rs [hunk 2]
```diff
@@ -50,6 +62,12 @@ decl_module! {
 			<PreKeyBundles<T>>::mutate((sender, device_id), |current_pkbs| current_pkbs.extend(pkbs));
 		}
 
+		/// Add a new PreKey bundle for a given user's device.
+		///
+		/// weight:
+		/// O(1) where n is the number of users and devices registered
```
Comment (jordy25519) [inline crml/sylo/src/e2ee.rs]: maybe out of date? `O(1) where n is the number of users and devices registered`

File: crml/sylo/src/groups/mod.rs [hunk 3]
```diff
@@ -140,6 +153,12 @@ decl_module! {
 			Ok(())
 		}
 
+		/// Leaves a group. If no one is left at the group, delete the group
+		///
+		/// weight:
+		/// O(n1) where n is the number of groups, and m is the number of members in that group
```
Comment (jordy25519) [inline crml/sylo/src/groups/mod.rs]: maybe out of date? `O(n1)`

File: crml/sylo/src/groups/mod.rs [hunk 4]
```diff
@@ -244,6 +281,12 @@ decl_module! {
 			Ok(())
 		}
 
+		/// Accept the invitation and add a user to the group
+		///
+		/// weight:
+		/// O(n) + O(m) where n is the number of groups, and m is the number of members in the group
```
Comment (jordy25519) [inline crml/sylo/src/groups/mod.rs]: `O(n + m)`

**Review by jordy25519 — COMMENTED** [review]
ty for this. Some comments might be out of date but otherwise lgtm

File: crml/sylo/src/e2ee.rs (hunk 1 above)
Comment (jordy25519) [reply crml/sylo/src/e2ee.rs]: \`\`\`\<comment\>\`\`\` is this triple \` an IDE config thing?

Comment (roy-powerfinance) [reply crml/sylo/src/e2ee.rs]: yep fixed

File: crml/sylo/src/e2ee.rs (hunk 2 above)
Comment (roy-powerfinance) [reply crml/sylo/src/e2ee.rs]: yep out of dated. this is now fixed

File: crml/sylo/src/groups/mod.rs (hunk 3 above)
Comment (roy-powerfinance) [reply crml/sylo/src/groups/mod.rs]: fixed

File: crml/sylo/src/groups/mod.rs (hunk 4 above)
Comment (roy-powerfinance) [reply crml/sylo/src/groups/mod.rs]: fixed

**Review by jordy25519 — APPROVED** [review]
👍

**Pull request merged**